src/components/search-input/index.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from 'react'
+
+import CustomIcon from '@/components/custom-icon'
+import { debounce } from '@/common/utils'
+
+import './index.scss'
+
+function SearchInput({ onChange }) {
+  const [focused, setFocused] = useState(false)
+
+  const onFocus = () => setFocused(true)
+
+  const onBlur = () => setFocused(false)
+
+  return (
+    <div className={`search-input ${focused ? 'search-input__focused' : ''}`}>
+      <CustomIcon className="search-input__prefix" name="search" size={18} />
+      <input
+        className="search-input__input"
+        onFocus={onFocus}
+        onBlur={onBlur}
+        onChange={debounce(onChange, 200, { trailing: true })}
```

Comment:
Is this debounce func working? Kinda seems like it's not working.

Reply:
Thanks, I forgot that the function will be regenerated every time in render😅, so debounce will not take effect. We need to keep the reference unchanged.